src/server/rdb_load.cc

```diff
@@ -2115,11 +2115,13 @@ std::error_code RdbLoaderBase::EnsureRead(size_t min_sz) {
     return std::error_code{};
   if (mem_buf_->InputLen() >= min_sz)
     return std::error_code{};
-  return EnsureReadInternal(min_sz);
+  // We need to include what we already read inside Input buffer. Otherwise we might expect to read
+  // more than the minimum
+  return EnsureReadInternal(min_sz - mem_buf_->InputLen());
 }
 
 error_code RdbLoaderBase::EnsureReadInternal(size_t min_sz) {
-  DCHECK_LT(mem_buf_->InputLen(), min_sz);
+  DCHECK_LT(mem_buf_->InputLen(), min_sz + mem_buf_->InputLen());
```

Comment:
this is awalys true for unsigned integers 🙂

Reply:
it's just that the meaning of the variable changed now, it's not min_sz (as min required size), but min_to_read (as to minimally should be read), so the fix could have also been applied here